collection.go

```diff
@@ -225,8 +225,24 @@ func (cs *CollectionSpec) LoadAndAssign(to interface{}, opts *CollectionOptions)
 		}
 	}
 
+	validate := func(assigned map[interface{}]structField) error {
```

Comment:
I find closure arguments to functions always a bit unergonomic, since they end up not being very flexible. In this case it also exposes internal state via `assigned` which makes it harder to change the implementation later.

Why not have a `var assignedMaps map[string]bool` which you populate from the `reflect.TypeOf((*Map)(nil))` arm in `getValue`? Then you can do the same check without the need for a closure, after `assignValues` has succeeded.

Reply:
After discussing offline, assigning to the receiver struct before throwing errors is not really an issue, as all assigned objects are closed in case of an error. Will revert a few changes accordingly.